compiler/src/dotty/tools/dotc/typer/ErrorReporting.scala

```diff
@@ -28,11 +28,14 @@ object ErrorReporting {
 
   def cyclicErrorMsg(ex: CyclicReference)(implicit ctx: Context) = {
     val cycleSym = ex.denot.symbol
-    def errorMsg(msg: String, cx: Context): Message =
+    def errorMsg(msg: Message, cx: Context): Message =
       if (cx.mode is Mode.InferringReturnType) {
         cx.tree match {
           case tree: untpd.DefDef if !tree.tpt.typeOpt.exists =>
-            OverloadedOrRecursiveMethodNeedsResultType(tree.name)
+            // TODO: analysis if tree is an overloaded method (or directly recursive)
```

Comment:
That is what I get upon calling `member(cycleSym.name)` - not sure how the code gets there.

[error] Test dotty.tools.dotc.reporting.ErrorMessagesTests.recursiveMethodNeedsReturnType failed: java.lang.AssertionError: assertion failed, took 0.01 sec
[error]     at scala.Predef$.assert(Predef.scala:151)
[error]     at dotty.tools.dotc.ast.Trees$Tree.withType(Trees.scala:116)
[error]     at dotty.tools.dotc.typer.ErrorReporting$.errorTree(ErrorReporting.scala:22)
[error]     at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1584)
[error]     at dotty.tools.dotc.typer.Typer$$anonfun$typed$2.apply(Typer.scala:1582)
[error]     at dotty.tools.dotc.reporting.Reporting$class.traceIndented(Reporter.scala:136)
[error]     at dotty.tools.dotc.core.Contexts$Context.traceIndented(Contexts.scala:57)
[error]     at dotty.tools.dotc.typer.Typer.typed(Typer.scala:1582)
...